athena-federation-sdk/src/test/java/com/amazonaws/athena/connector/lambda/serde/ObjectMapperUtil.java

```diff
@@ -31,17 +39,64 @@
 
 public class ObjectMapperUtil
 {
+    private static final Logger logger = LoggerFactory.getLogger(ObjectMapperUtil.class);
+
+    private static final JsonFactory jsonFactory = new JsonFactory();
+    private static final V24SerDeProvider serDeProvider = new V24SerDeProvider();
+
+    private static final boolean failOnObjectMapperChecks = false;
+
     private ObjectMapperUtil() {}
 
-    public static <T> void assertSerialization(Object object, Class<T> clazz)
+    public static <T> void assertSerialization(Object object, BlockAllocator allocator)
```

Comment:
The reason why this class creates its own BlockAllocator is that ApacheArrow is zero copy and if you attempt to read then write something to/from the same underlying Allocator... ApacheArrow doesn't actually do anything in some cases because it attempts to be zero-copy. In other cases this means that when you write into what you thought was a copy you are actually changing the originally too. Was there a reason why you add to change this to accept a BlockAllocator? If so, that might be a sign we broke something.